File: ForcWebApi/Infrastructure/Models/UserTarget.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace ForcWebApi.Infrastructure.Entities
{
    /// <summary>
    /// Цель пользователя
    /// </summary>
    public class UserTarget : BaseEntity<Guid>
    {
        /// <summary>
        /// Пользователь
        /// </summary>
        public Guid UserId { get; set; }

        /// <summary>
        /// Актуальность
        /// </summary>
        public bool Relevance { get; set; }

        /// <summary>
        /// Дата начала
        /// </summary>
        public DateTime DateStart { get; set; }

        /// <summary>
        /// Дата окончания
        /// </summary>
        public DateTime DateFinish { get; set; }

        /// <summary>
        /// Текущий вес тела
        /// </summary>
        public double CurrentBodyWeight { get; set; }

        /// <summary>
        /// Целевой вес тела
        /// </summary>
        public double TargetBodyWeight { get; set; }

        /// <summary>
        /// ID дневной нормы в соответствии с актуальной целью
        /// </summary>
        public Guid DailyRateId { get; set; }

        /// <summary>
        /// Дневная норма в соответствии с актуальной целью
        /// </summary>
        public DailyRate DailyRate { get; set; }
    }
}

File: Forc.WebApp/src/app/modules/foodDiaryModule/userProfile/userPage/userTargets/userTarget.ts
import { Guid } from 'guid-typescript';
import { BaseEntity } from 'src/app/modules/shared/models/baseEntity';
import { DailyRate } from '../dailyRate/dailyRate';

/* Цель пользователя */
export class UserTarget extends BaseEntity {
	/* ID пользователя */
	userId: Guid;

	/* Актуальность */
	relevance: boolean;

	/* Дата начала */
	dateStart: Date;

	/* Дата окончания */
	dateFinish: Date;

	/* Текущий вес тела */
	currentBodyWeight: number;

	/* Целевой вес тела */
	targetBodyWeight: number;

	/* ID дневной нормы в соответствии с актуальной целью */
	dailyRateId?: Guid;

	/* Дневная норма в соответствии с актуальной целью */
	dailyRate?: DailyRate;
}